src/data/list/basic.lean

```diff
@@ -474,6 +474,35 @@ by cases n; refl
 @[simp] theorem join_repeat_nil (n : ℕ) : join (repeat [] n) = @nil α :=
 by induction n; [refl, simp only [*, repeat, join, append_nil]]
 
+lemma repeat_inj_left {a b : α} {n : ℕ} (npos : 0 < n)
```

Comment:
It looks like `*_left_inj` and `*_right_inj` are more common in the library, and the example at our [naming conventions page](https://leanprover-community.github.io/contribute/naming.html#general-conventions) is `sub_right_inj`, so I'd recommend switching the naming over to that.

This convention might be confusing given all the other things named with `left` and `right` at the very end, but that's something we should discuss on Zulip if someone feels strongly about it.

Reply:
I renamed lemmas, added `function.injective` versions, and golfed the proofs.